src/controller/python/chip/ChipBleBase.py

```diff
@@ -0,0 +1,136 @@
+        return
+
+    @abc.abstractmethod
+    def scan_connect(self, line):
+        """ API to perform both scan and connect operations in one call."""
+        return
+
+    @abc.abstractmethod
+    def disconnect(self):
+        """ API to initiate BLE disconnect procedure."""
+        return
+
+    @abc.abstractmethod
+    def connect(self, identifier):
+        """ API to initiate BLE connection to peripheral device whose identifier == identifier."""
+        return
+
+    @abc.abstractmethod
+    def CloseBle(self, connObj):
+        """ Called by Chip to close the BLE connection."""
+        return
+
+    @abc.abstractmethod
+    def GetBleEvent(self):
+        """ Called by ChipDeviceMgr.py on behalf of Chip to retrieve a queued message."""
+        return
+
+    @abc.abstractmethod
+    def devMgrCB(self):
+        """ A callback used by ChipDeviceMgr.py to drive the runloop while the
+            main thread waits for the Chip thread to complete its operation."""
+        return
+
+    @abc.abstractmethod
+    def readlineCB(self):
+        """ A callback used by readline to drive the runloop while the main thread
+            waits for commandline input from the user."""
+        return
+
+    @abc.abstractmethod
+    def setInputHook(self, hookFunc):
+        """Set the PyOS_InputHook to call the specific function."""
+        return
+
+    @abc.abstractmethod
+    def runLoopUntil(self, should_tuple):
+        """ helper function to drive runloop until an expected event is received or
+            the timeout expires."""
+        return
+
+    def Usage(self, cmd):
+        if cmd is None:
+            return
+
+        line = "USAGE: "
+
+        if cmd == "scan":
+            line += "ble-scan [-t <timeout>] [<name>|<identifier>] [-q <quiet>]"
+        elif cmd == "scan-connect":
+            line += "ble-scan-connect [-t <timeout>] <name> [-q <quiet>]"
+
+        self.logger.info(line)
+
+    def ParseInputLine(self, line, cmd=None):
+        args = shlex.split(line)
+        optParser = optparse.OptionParser(usage=optparse.SUPPRESS_USAGE)
+
+        if cmd == "scan" or cmd == "scan-connect":
+            optParser.add_option("-t", "--timeout", action="store", dest="timeout", type="float", default=10.0)
+            optParser.add_option("-q", "--quiet", action="store_true", dest="quiet")
+
+        try:
+            (options, remainingArgs) = optParser.parse_args(args)
+        except SystemExit:
+            self.Usage(cmd)
+            return None
+
+        if cmd is None:
+            return remainingArgs
+
+        if len(remainingArgs) > 1:
+            self.Usage(cmd)
+            return None
+
+        name = None
+
+        if len(remainingArgs):
+            name = str(remainingArgs[0])
+        elif cmd == "scan-connect":
+            self.Usage(cmd)
+            return None
+
+        return (options.timeout, options.quiet, name)
```

Comment:
nit: add final newline so that it shows better in diffs.

Could we auto-format these files?

Reply:
Done, just autoformat them. thanks